### PR TITLE
Limit connection reuse

### PR DESCRIPTION
**Pull request by skontar**

### CODE REVIEW

File: collectors/jiraffe/constants.py [hunk 1]
```diff
@@ -7,3 +7,6 @@
 
 # Jira datetime format string
 JIRA_DT_FMT = "%Y-%m-%d %H:%M"
+
+# Maximum age of connection to Jira in seconds, recommended value is 60
+JIRA_MAX_CONNECTION_AGE = get_env("JIRA_MAX_CONNECTION_AGE")
```
Comment (osoukup) [inline collectors/jiraffe/constants.py:12]: Same comment with the explicit default applies here.

**Review by osoukup — APPROVED** [review]
LGTM

File: collectors/bzimport/constants.py [hunk 2]
```diff
@@ -26,3 +26,6 @@
 BZ_DT_FMT_HISTORY = "%Y-%m-%dT%H:%M:%SZ"
 
 FLAW_PLACEHOLDER_KEYWORD = "Tracking"
+
+# Maximum age of connection to Bugzilla in seconds, recommended value is 60
+BZ_MAX_CONNECTION_AGE = get_env("BZ_MAX_CONNECTION_AGE")
```
[the remark this answers is not in the document]
Comment (skontar) [reply collectors/bzimport/constants.py:31]: Default is `None` so it is disabled by default on purpose.

1. It does not mess with tests where connection aging would add non-determinism
2. It is probably not needed as I have not seen any `ConnectionResetError` failures on stage when the branch was deployed. It seems that decoupling `Queriers` from `Collectors` has the biggest impact.  However, it gives us ability to enable it any time if we need it (maybe production may behave differently, maybe if we ever re-populate whole database etc.)

[the remark this answers is not in the document]
Comment (osoukup) [reply collectors/bzimport/constants.py:31]: My bad. When writing this comment I forgot that it is conditional and thought it must be always set to some value which I thought is missing by default. Never mind.

**Review by Elkasitu — APPROVED** [review]
Could we override `__init_subclass__` in the various `Querier` classes to prevent them from being subclassed by raising an error?

**Comment by Elkasitu** [comment]
> @Elkasitu not exactly. I think you do not want a subclass of a `Collector` and `Querier` together. But there is for example `JiraTaskmanQuerier`, `TrackerJiraSaver`, `BugzillaSaver` which subclass `Querier` classes.

Good point, maybe anything that has celery `Task` in its bases? Or just documentation

**Comment by Elkasitu** [comment]
> @Elkasitu do you think it can be done? Like if you subclass `Querier`, check if it also subclasses celery `Task`? Maybe in `Querier` `__init__`?

```python
>>> class Task:
...     pass
... 
>>> class Querier:
...     def __init_subclass__(cls):
...             if issubclass(cls, Task):
...                     raise TypeError("Class Querier cannot be subclassed by subclasses of Task")
...             super().__init_subclass__()
... 
>>> class Collector(Task, Querier):
...     pass
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 4, in __init_subclass__
TypeError: Class Querier cannot be subclassed by subclasses of Task
>>> class SuperQuerier(Querier):
...     pass
... 
```

Just a PoC though, there might be other ways of doing it, but if done in `__init__` it will be raised at run-time, whereas with `__init_subclass__` it will be raised at "compile-time" (when the class declaration is read).

**Pull request merged**